### PR TITLE
[mlir][vector] Hoist uniform scalar loop code after scf.for distribution

### PR DESCRIPTION
**Pull request by qedawkins**
After propagation of `vector.warp_execute_on_lane_0` through `scf.for`, uniform operations like those on the loop iterators can now be hoisted out of the inner warp op.

### CODE REVIEW

**Review by kuhar — APPROVED** [review]
LGTM but you may want to wait for one more approval. I also tweaked the title slightly.

**Review by antiagainst — APPROVED** [review]

**Pull request merged**
